### PR TITLE
ignore invalid trace indices in restyle and update

### LINKED ISSUE #3113: `Plotly.restyle` and (`Plotly.update`) throws error if trace indice is invalid or out of bound
`restyle` (and `update`) should not throw an error if you give invalid trace indices, it should just toss them (this behavior was requested by @alexcjohnson)!

Here's a codepen reproducing the issue: https://codepen.io/anon/pen/RexqKO

### PR DESCRIPTION
**Pull request by antoinerg**
Fixes #3113

### CODE REVIEW

File: src/plot_api/helpers.js [hunk 1]
```diff
@@ -490,6 +490,9 @@ exports.coerceTraceIndices = function(gd, traceIndices) {
     else if(!Array.isArray(traceIndices) || !traceIndices.length) {
         return gd.data.map(function(_, i) { return i; });
     }
+    else if(Array.isArray(traceIndices)) {
+        return traceIndices.filter(function(i) {return i < gd.data.length;});
+    }
```
Comment (antoinerg) [inline src/plot_api/helpers.js]: Should I do a `Lib.warn('indice is not number or is out of bound')?

Comment (antoinerg) [reply src/plot_api/helpers.js]: Hooray for `lib.isIndex`. I really should spend some time to find out all the functions we have in `lib`!

**Comment by alexcjohnson** [comment]
Thanks for the quick fix @antoinerg ! 🚀 Very nice tests.
💃

**Comment by etpinard** [comment]
:sos: delete branch alert :sos:

**Pull request merged**